src/sql/workbench/contrib/query/browser/fileQueryEditorInput.ts

```diff
@@ -86,4 +88,29 @@ export class FileQueryEditorInput extends QueryEditorInput {
 	public override rename(group: GroupIdentifier, target: URI): IMoveResult {
 		return this.text.rename(group, target);
 	}
+
+	override async saveAs(group: GroupIdentifier, options?: ISaveOptions): Promise<IEditorInput | undefined> {
+		let newEditorInput = await this.text.saveAs(group, options);
+		let newUri = newEditorInput.resource.toString(true);
+		if (newUri === this.uri) {
+			// URI is the same location, no need to change URI for the query in services, just return input.
+			return newEditorInput;
+		}
+		else {
+			// URI is different, need to update URI for the query in services to ensure we can keep the current results/view state
+			// without resetting and creating a brand new query.
+			try {
+				await this.changeConnectionUri(newUri);
+				this._results.uri = newUri;
+				// Create a new FileQueryEditorInput with current results and state in order to trigger a rename for editor tab name.
+				// (Tab name won't refresh automatically if current input is reused directly)
+				let newFileQueryInput = this.instantiationService.createInstance(FileQueryEditorInput, '', (newEditorInput as FileEditorInput), this.results);
+				newFileQueryInput.state.setState(this.state);
+				return newFileQueryInput;
+			}
+			catch (error) {
+				return newEditorInput;
```

Comment:
Comment explaining why we're returning the plain file input here and what that means for the editor.

But in general I don't like that we have something like this. Errors shouldn't be expected and so having fallback stuff like this seems to imply we think it would happen often enough we have to account for that. It'd be good to figure out why/when this case would be hit and try to fix those scenarios in the long term.

Reply:
I agree that it shouldn't be, and in fact it works without having the error checks altogether, but there needs to be a delay or some way to check that the file has updated properly before running the grid refresh on the file and this will require another PR and some time to investigate.